### PR TITLE
fix scaladoc for Eval methods Unit, True, False, Zero, One

### PR DESCRIPTION
**Pull request by sullivan-**

### CODE REVIEW

**Comment by johnynek** [comment]
👍

File: core/src/main/scala/cats/Eval.scala [hunk 1]
```diff
@@ -206,15 +206,43 @@ object Eval extends EvalInstances {
     new Eval.Call[A](a _) {}
 
   /**
-   * Static Eval instances for some common values.
+   * Static Eval instances for common value `Unit`.
```
Comment (kailuowang) [inline core/src/main/scala/cats/Eval.scala]: `instances` -> `instance`

Comment (sullivan-) [inline core/src/main/scala/cats/Eval.scala]: oops! fixed it

**Comment by kailuowang** [comment]
Thanks very much！

**Pull request merged**